### ISSUE TITLE
transformer.encoder.forward seems to have uncorrect docs

### ISSUE THREAD

**Issue by GodEase** [espnet/espnet#4567]
**Describe the bug**
While the DOCS say, this forward function needs the input masks to be a tensor with shape (#batch, time), it actually needs a mask tensor with shape (#batch, 1, time) for multi-head self attention.

**Basic environments:**
 - OS information: Linux 5.4.0-100-generic #113~18.04.1-Ubuntu SMP Mon Feb 7 15:02:59 UTC 2022 x86_64
 - python version: `3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:58:50)  [GCC 10.3.0]`
 - espnet version: `espnet 202207`
 - Git hash: `4e4f327bea950e377a61faa29ad366a3d2b0fd81`
   - Commit date: `Thu Jul 14 16:18:22 2022 +0800`
 - pytorch version: `pytorch 1.12.0+cu113`

**Task information:**
 - Task: TTS

**To Reproduce**
```
encoder = TransformerEncoder(idim=384)
encoder(xs=torch.randn(2, 10, 384), masks=torch.zeros(2, 1, 10)) # this works fine
encoder(xs=torch.randn(2, 10, 384), masks=torch.zeros(2, 10)) # this will raise IndexError

# further more, in fastspeech2 which use _source_mask to get the masks, returns a (#batch, 1, time) tensor
from espnet2.tts.fastspeech2 import FastSpeech2
fs2 = FastSpeech2(idim=384, odim=80)
fs2._source_mask(ilens=[10,10]).size()  # torch.Size([2, 1, 10])
```
